pennylane/ftqc/parametric_midmeasure.py

```diff
@@ -209,39 +200,56 @@ def circuit(x):
     Applying the transform inserts the relevant gates before the measurement to allow
     measurements to be in the Z basis, so the original circuit
 
-    >>> print(qml.draw(circuit, level=0)([np.pi/4, np.pi/4]))
+    >>> print(qml.draw(circuit, level=0)([np.pi/4, np.pi]))
     0: ──RY(0.79)──┤↗ˣʸ(3.14)├─┤  <Z>
 
     becomes
 
-    >>> print(qml.draw(circuit)([np.pi/4, np.pi/4]))
-    0: ──RY(0.79)──U(M0)──┤↗ᶻʸ(0.0)├─┤  <Z>
-    M0 =
-    [[ 0.70710678+0.j   0.5       -0.5j]
-     [ 0.70710678+0.j  -0.5       +0.5j]]
-
+    >>> print(qml.draw(circuit)([np.pi/4, np.pi]))
+    ──RY(0.79)──Rϕ(-3.14)──H──┤↗├─┤  <Z>
     """
 
     new_operations = []
+    mps_mapping = {}
 
     for op in tape.operations:
-        # ToDo: maybe this in-place modification is a bad idea, but
-        #  its also weird to diagonalize it and not update it
         if isinstance(op, ParametricMidMeasureMP):
+
+            # add diagonalizing gates to tape
             diag_gates = op.diagonalizing_gates()
-            op.angle = 0
-            op.plane = "ZY"
             new_operations.extend(diag_gates)
+
+            # add comuptational basis MCM to tape
+            new_mp = MidMeasureMP(op.wires, reset=op.reset, postselect=op.postselect, id=op.id)
+            new_operations.append(new_mp)
+
+            # track mapping from original to computational basis MCMs
+            mps_mapping[op] = new_mp
+
         elif isinstance(op, qml.ops.Conditional) and isinstance(op.base, ParametricMidMeasureMP):
+
+            # from MCM mapping, map any MCMs in the condition if needed
+            processing_fn = op.meas_val.processing_fn
+            mps = [mps_mapping.get(op, op) for op in op.meas_val.measurements]
+            expr = MeasurementValue(mps, processing_fn=processing_fn)
+
+            # add conditional diagonalizing gates to the tape
```

Comment:
I'm unable to create a suggestion, but this whole branch also needs to use the `qml.QueuingManager.stop_recording()` context manager when creating new ops (new `Conditional`s, diagonalizing gates, and new `MidMeasureMP`s)

Reply:
Aaah, thanks! Should be good for another look now :)

Reply:
No, wait, I'm having a problem in the follow-up PR and I just understood why. I'm only diagonalizing conditionals in the `true_fn` and `false_fn`. I'm not diagonalizing them if they appear in the condition of cond 😅 . I will fix this tomorrow and _then_ it should be good for another look.

Reply:
That wasn't exactly the problem (it was that I was only fixing the mcm references on the conditionals that applied parametrized MCMs), but its fixed now.